lib/srv/monitor.go

```diff
@@ -405,15 +405,26 @@ func (w *Monitor) disconnectClientOnExpiredCert() {
 	w.disconnectClient(reason)
 }
 
+type canCloseWithCause interface {
```

Comment:
nit: can we ranme. it like withCauseCloser ? The `canCloseWithCause` should like a function name that returns bool